---
title: "Similarity between copy number profiles"
author: Astrid Deschênes and Pascal Belleau
output:
  BiocStyle::html_document:
    toc: true
bibliography: CNVMetrics.bibtex
vignette: >
  %\VignetteIndexEntry{Copy number variant metrics}
  %\VignettePackage{CNVMetrics}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r style, echo = FALSE, results = 'hide', warning=FALSE, message=FALSE}
BiocStyle::markdown()

suppressPackageStartupMessages({
  library(knitr)
  library(GenomeInfoDb)
  library(GenomicRanges)
  library(CNVMetrics)
})
```

<br />
**Package**: `r Rpackage("CNVMetrics")`<br />
**Authors**: `r packageDescription("CNVMetrics")[["Author"]]`<br />
**Version**: `r packageDescription("CNVMetrics")$Version`<br />
**Compiled date**: `r Sys.Date()`<br />
**License**: `r packageDescription("CNVMetrics")[["License"]]`<br />


# Licensing 

The `r Githubpkg("adeschen/CNVMetrics")` package and the underlying 
`r Githubpkg("adeschen/CNVMetrics")` code are distributed under the 
Artistic license 2.0. You are free to use and redistribute this software. 

# Introduction

Copy number variation (CNV) includes multiplication and deletion of DNA
segment (> 1 kb). Copy number variations have been shown to be associated 
with a wide spectrum of pathological conditions and complex traits, such 
as developmental neuropsychiatric disorders [@Hiroi2013] and especially 
cancer [@Stratton2009]. 

CNV are usually reported, for each sample, as genomic regions that are 
duplicated or deleted with respect to a reference. Those regions are denoted 
as _CNV calls_. The level of amplification or deletion can also be reported, 
usually in log2 ratio values or normalized read depth [@Zhao2013]. As an 
example, the Figure 1 shows the copy number profiles from sequencing data of 
two mouse pancreatic organoids [@Oni2020], calculated with 
`r Githubpkg("KrasnitzLab/CNprep")` [@Belleau2020] and plot with 
`r Biocpkg("gtrellis")` [@Gu2016a].


```{r graphCNVpro, echo = FALSE, fig.align="center", fig.cap="Copy number profiles of two mouse metastatic pancreatic organoids (M10 and M30).", out.width = '95%'}
knitr::include_graphics("CNV_mM30_mM10_v03_Feb_08_2021_small.jpeg")
```


While visual representation are a practical way to qualitatively compare copy 
number profiles, metrics are useful statistical tools for quantitatively 
measuring similarity and dissimilarity between profiles. Similarity metrics can
be used to compare independent samples but also related samples such as 
copy number profiles from bulk cancer tissues and paired organoids generated 
from those.

The `r Githubpkg("adeschen/CNVMetrics")` package calculates metrics to 
estimate the level of similarity between copy number profiles. Some metrics 
are calculated using the _CNV calls_ (amplification/deletion status) while 
others are based on the level of amplification/deletion. Finally, a 
visualization tool is provided to explore resulting metrics.

# Installation

To install the latest version accessible on the 
[CNVMetrics Github Website](https://github.com/adeschen/CNVMetrics/releases "CNVMetrics Github Site"), the `r CRANpkg("devtools")` package is required.

```{r installDemo01, eval=FALSE}
## Load required package
library(devtools)

## Install the latest version of CNVMetrics
devtools::install_github('adeschen/CNVMetrics')
```

It is also possible to install an official release. The list of available 
releases is posted on the [CNVMetrics Release Website](https://github.com/adeschen/CNVMetrics/releases "CNVMetrics Release Site").


```{r installDemo02, eval=FALSE}
## Load required package
library(devtools)

## Install the version v0.1.2 of CNVMetrics
## using 'ref' parameter
devtools::install_github('adeschen/CNVMetrics', ref = "v0.1.2")
```

# Workflow for metrics calculated using _CNV calls_

The following figure gives an overview of the capabilities of  `r Githubpkg("adeschen/CNVMetrics")` to calculate metrics using the 
_CNV calls_ (amplification/deletion status):


The key functions for each step are:

 Step                   | Function                    
----------------------- | ---------------------------------------------
 Data Importation       | `GenomicRanges::makeGRangesListFromDataFrame()`
 Metric Calculation     | `calculateOverlapMetric()`   
 Metric Visualization   | `plotOverlapMetric()`
 
The `package::function()` notation is used for functions from other packages.

## Data Input - Copy number file

_CNV calls_ are represented as segments with a copy number state. The state 
be general, such as "amplification", "deletion" or "neutral", or more 
specific such as TODO.

A basic five-column input file containing genomic position 
(chromosome, start, end), sample identification and CNV calls is required. 
All samples that need to be analyzed together have to be present in the file. 

A column named **state** is required. In this column, the amplified and 
deleted segments must be assigned those values:

* **<span style="color:darkred">AMPLIFICATION</span>**
* **<span style="color:darkred">DELETION</span>**

Segments with other state values can be present in the file. However, those 
segments won't be retain for the calculation of the metrics.

```{r figureCNVFile, echo = FALSE, fig.align="center", fig.cap="Example of a copy number file containing CNV calls.", out.width = '95%'}
knitr::include_graphics("Input_CNfile_2_small.jpg")
```


## Data Importation - GRangesList

The input format for the copy number information, as needed by the 
`calculateOverlapMetric()` function, is a `GRangesList` object.

The easiest way to generate a `GRangesList` object is to first load the 
copy number information into an R `data.frame` and then, use the 
`GenomicRanges::makeGRangesListFromDataFrame()` function to convert them 
to a `GRangesList`.

## Metric Calculation

The calculation of the similarity metrics is done with the 
`calculateOverlapMetric()` function. 

## Metric Visualization  

TODO

# Metrics using the _CNV calls_

This survey represents the similarity measures that are implemented in 
`r Githubpkg("adeschen/CNVMetrics")` package. Those metrics are calculated 
using the _CNV calls_. The size of the amplified/deleted regions as well as 
the size of the overlapping of regions are always in base paired. 

## Sørensen


The Sørensen coefficient [@Sorensen48] is calculated by dividing twice the  
size of the intersection by the sum of the size of the two sets:

\begin{equation}
  \frac{2\times  \left| X \cap Y \right| }{\left| X \right| + \left| Y \right|}
  (\#eq:sorensen)
\end{equation}    


## Szymkiewicz–Simpson

The Szymkiewicz–Simpson coefficient [@Vijaymeena2016], also known as the 
overlap coefficient, is calculated by dividing the size of the intersection 
by the smaller of the size of the two sets:

\begin{equation}
  \frac{\left| X \cap Y \right| }{min \left(\left| X \right| , \left| Y \right|\right)}
  (\#eq:szymkiewicz)
\end{equation}    

If set **X** is a subset of **Y** or vice versa, the overlap coefficient 
value is 1. 


<!---
![Figure 2. Copy number profile of a mouse metastatic pancreatic organoid.](overlap_demo2.jpg "Copy number profile"){ width=100% }
-->


# Metrics using the level of amplification/deletion

This survey represents the similarity measures that are implemented in 
`r Githubpkg("adeschen/CNVMetrics")` package.

TODO




# Inputs

## Chromosomes information 

The chromosomes information is mandatory. It ensures that only the selected
chromosomes are used in the analysis. As not all CNV software use the same
reference (reference with alternative regions or not), it ensure that
all samples will be compared using the same reference. We strongly encourage 
using only the primary chromosomes for this analaysis.

The chromosomes information is contained inside a `Seqinfo` object. The 
information from some UCSC genomes can be fetched automatically using
the `r Biocpkg("GenomeInfoDb")` package. 

```{r knownGenome, collapse=TRUE, eval=FALSE, message=FALSE, warning=FALSE}
### Import needed library that contains the Seqinfo method
library(GenomeInfoDb)

### Get the information for Human genome version GRCh38
hg38Info <- Seqinfo(genome="hg38")

### Subset the object to keep only the analyzed chromosomes
### Chromosomes X and Y are often removed from subsequent analyses
seqlevels(hg38Info) <- paste0("chr", 1:22)
```

A `Seqinfo` object can also be created using the chromosomes information 
specific to the analyzed genome. The `r Biocpkg("GenomeInfoDb")` package is
required.

```{r newGenome, collapse=FALSE}
### Import needed library that contains the Seqinfo method
library(GenomeInfoDb)

### Create an Seqinfo Object
chrInfo <- Seqinfo(seqnames=c("chr1", "chr2", "chr3"),
            seqlengths=c(10000, 20000, 1500), isCircular=c(FALSE, FALSE, FALSE),
            genome="Alien")
```

```{r deleteChr, echo=FALSE, message=FALSE}
if (exists("chrInfo", inherits = FALSE)) rm(chrInfo)
if (exists("hg19Subset", inherits = FALSE)) rm(hg19Subset)
if (exists("hg19Info", inherits = FALSE)) rm(hg19Info)
```



```{r demo}
#a<-prepareInformation(segDirectory = "inst/extdata/", bedExclusionFile = "inst/extdata/exclusion.bed", segmentWithHeader = FALSE, chrInfo = hg38Info)

```


# Session info

Here is the output of `sessionInfo()` on the system on which this document was 
compiled:

```{r sessionInfo, echo=FALSE}
sessionInfo()
```


# References


