<!DOCTYPE html>
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="pandoc">
<meta http-equiv="X-UA-Compatible" content="IE=EDGE">
<meta name="author" content="Astrid Deschênes, Pascal Belleau and Alexander Krasnitz">
<meta name="date" content="2021-07-01">
<title>Quantifying similarity between copy number profiles</title>
<script src="CNVMetrics_files/header-attrs-2.6/header-attrs.js"></script><script src="CNVMetrics_files/jquery-1.11.3/jquery.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1">
<link href="CNVMetrics_files/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet">
<script src="CNVMetrics_files/bootstrap-3.3.5/js/bootstrap.min.js"></script><script src="CNVMetrics_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script><script src="CNVMetrics_files/bootstrap-3.3.5/shim/respond.min.js"></script><script src="CNVMetrics_files/navigation-1.1/tabsets.js"></script><style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>
<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script><style type="text/css">

</style>
<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>
<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>
<link rel="stylesheet" href="/__w/_temp/Library/BiocStyle/resources/html/bioconductor.css" type="text/css">
<style type="text/css">
.main-container {
  max-width: 828px;
  margin-left: auto;
  margin-right: auto;
}

img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>
<!-- tabsets --><style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>
<!-- code folding --><script>
function toggle_visibility(id1) {
  var e = document.getElementById(id1);
  e.style.display = ((e.style.display!="none") ? "none" : "block");
}
</script>
</head>
<body>


<div class="container-fluid main-container">




<div class="fluid-row" id="header">



<h1 class="hasAnchor">
<a href="#header" class="anchor"></a>Quantifying similarity between copy number profiles</h1>
<p class="author-name">Astrid Deschênes, Pascal Belleau and Alexander Krasnitz</p>
<h4 class="date">1 July 2021</h4>

</div>

<h1>Contents</h1>
<div id="TOC">
<ul>
<li><a href="#licensing"><span class="toc-section-number">1</span> Licensing</a></li>
<li><a href="#introduction"><span class="toc-section-number">2</span> Introduction</a></li>
<li><a href="#installation"><span class="toc-section-number">3</span> Installation</a></li>
<li>
<a href="#workflow-for-metrics-calculated-using-cnv-status-calls"><span class="toc-section-number">4</span> Workflow for metrics calculated using <em>CNV status calls</em></a>
<ul>
<li><a href="#data-input---copy-number-file"><span class="toc-section-number">4.1</span> Data Input - Copy number file</a></li>
<li><a href="#data-importation---grangeslist"><span class="toc-section-number">4.2</span> Data Importation - GRangesList</a></li>
<li><a href="#metric-calculation"><span class="toc-section-number">4.3</span> Metric Calculation</a></li>
<li><a href="#metric-visualization"><span class="toc-section-number">4.4</span> Metric Visualization</a></li>
</ul>
</li>
<li>
<a href="#metrics-using-the-cnv-status-calls"><span class="toc-section-number">5</span> Metrics using the <em>CNV status calls</em></a>
<ul>
<li><a href="#s%C3%B8rensen"><span class="toc-section-number">5.1</span> Sørensen</a></li>
<li><a href="#szymkiewiczsimpson"><span class="toc-section-number">5.2</span> Szymkiewicz–Simpson</a></li>
<li><a href="#jaccard"><span class="toc-section-number">5.3</span> Jaccard</a></li>
</ul>
</li>
<li>
<a href="#metrics-using-the-level-of-amplificationdeletion"><span class="toc-section-number">6</span> Metrics using the level of amplification/deletion</a>
<ul>
<li><a href="#weighted-euclidean-distance"><span class="toc-section-number">6.1</span> Weighted Euclidean Distance</a></li>
</ul>
</li>
<li>
<a href="#supplementary-information"><span class="toc-section-number">7</span> Supplementary information</a>
<ul>
<li><a href="#creating-your-own-grangeslist"><span class="toc-section-number">7.1</span> Creating your own GRangesList</a></li>
</ul>
</li>
<li><a href="#session-info"><span class="toc-section-number">8</span> Session info</a></li>
<li><a href="#references">References</a></li>
</ul>
</div>

<p><br><strong>Package</strong>: <em>CNVMetrics</em><br><strong>Authors</strong>: Astrid Deschênes [aut, cre] (<a href="https://orcid.org/0000-0001-7846-6749" class="uri">https://orcid.org/0000-0001-7846-6749</a>),
Pascal Belleau [aut] (<a href="https://orcid.org/0000-0002-0802-1071" class="uri">https://orcid.org/0000-0002-0802-1071</a>),
Alexander Krasnitz [aut]<br><strong>Version</strong>: 0.1.4<br><strong>Compiled date</strong>: 2021-07-01<br><strong>License</strong>: Artistic-2.0<br></p>
<div id="licensing" class="section level1" number="1">
<h1 class="hasAnchor">
<a href="#licensing" class="anchor"></a><span class="header-section-number">1</span> Licensing</h1>
<p>The <em><a href="https://github.com/KrasnitzLab/CNVMetrics">CNVMetrics</a></em> package and the underlying
<em><a href="https://github.com/KrasnitzLab/CNVMetrics">CNVMetrics</a></em> code are distributed under the
Artistic license 2.0. You are free to use and redistribute this software.</p>
</div>
<div id="introduction" class="section level1" number="2">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a><span class="header-section-number">2</span> Introduction</h1>
<p>Copy number variation (CNV) includes multiplication and deletion of DNA
segment. Copy number variations have been shown to be associated
with a wide spectrum of pathological conditions and complex traits, such
as developmental neuropsychiatric disorders <span class="citation">(Hiroi et al. 2013)</span> and especially
cancer <span class="citation">(Stratton, Campbell, and Futreal 2009)</span>.</p>
<p>CNVs are usually reported, for each sample, as genomic regions that are
duplicated or deleted with respect to a reference. Those regions are denoted
as <em>CNV status calls</em>. The level of amplification or deletion can also be
reported, usually in log2 ratio values or normalized read depth <span class="citation">(Zhao et al. 2013)</span>.
As an example, the Figure 1 shows the copy number profiles from sequencing
data of two mouse pancreatic organoids <span class="citation">(Oni et al. 2020)</span>, calculated with
<em><a href="https://github.com/KrasnitzLab/CNprep">CNprep</a></em> <span class="citation">(Belleau et al. 2020)</span> and plot with
<em><a href="https://bioconductor.org/packages/3.12/gtrellis">gtrellis</a></em> <span class="citation">(Gu, Eils, and Schlesner 2016)</span>.</p>
<div class="figure" style="text-align: center">
<span id="fig:graphCNVpro"></span>
<img src="CNV_mM30_mM10_v03_Feb_08_2021_small.jpeg" alt="Copy number profiles of two mouse metastatic pancreatic organoids (M10 and M30)." width="95%"><p class="caption">
Figure 1: <span class="caption-title">Copy number profiles of two mouse metastatic pancreatic organoids (M10 and M30)</span><br></p>
</div>
<p>While visual representation is a practical way to qualitatively compare copy
number profiles, metrics are useful statistical tools for quantitatively
measuring similarity and dissimilarity between profiles. Similarity metrics
can be employed to compare CNV profiles of genetically unrelated samples.
Moreover, those metrics can as well be put to use on samples with common
genetic background. As an example, a comparison between primary and metastatic
tumor CNV profiles may reveal genomic determinants of metastasis. Similarly,
patient-derived xenograft or organoid models of cancer are expected to
recapitulate CNV patterns of the tumor tissue of origin <span class="citation">(Gendoo et al. 2019)</span>.</p>
<p>The <em><a href="https://github.com/KrasnitzLab/CNVMetrics">CNVMetrics</a></em> package calculates metrics to
estimate the level of similarity between copy number profiles. Some metrics
are calculated using the <em>CNV status calls</em> (amplification/deletion status)
while others are based on the level of amplification/deletion. Finally, a
visualization tool is provided to explore resulting metrics in the form of
sample-to-sample heatmaps.</p>
</div>
<div id="installation" class="section level1" number="3">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a><span class="header-section-number">3</span> Installation</h1>
<p>To install the latest version accessible on the
<a href="https://github.com/KrasnitzLab/CNVMetrics" title="CNVMetrics Github Site">CNVMetrics Github Website</a>,
the <em><a href="https://CRAN.R-project.org/package=devtools">devtools</a></em> package is required.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Load required package</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/">devtools</a></span><span class="op">)</span>

<span class="co">## Install the latest version of CNVMetrics</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">'KrasnitzLab/CNVMetrics'</span><span class="op">)</span></code></pre></div>
<p>It is also possible to install an official release. The list of available
releases is posted on the <a href="https://github.com/KrasnitzLab/CNVMetrics/releases" title="CNVMetrics Release Site">CNVMetrics Release Website</a>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Load required package</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/">devtools</a></span><span class="op">)</span>

<span class="co">## Install the version v0.1.4 of CNVMetrics</span>
<span class="co">## using 'ref' parameter</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">'KrasnitzLab/CNVMetrics'</span>, ref <span class="op">=</span> <span class="st">"v0.1.4"</span><span class="op">)</span></code></pre></div>
</div>
<div id="workflow-for-metrics-calculated-using-cnv-status-calls" class="section level1" number="4">
<h1 class="hasAnchor">
<a href="#workflow-for-metrics-calculated-using-cnv-status-calls" class="anchor"></a><span class="header-section-number">4</span> Workflow for metrics calculated using <em>CNV status calls</em>
</h1>
<p>The following figure gives an overview of the capabilities of
<em><a href="https://github.com/KrasnitzLab/CNVMetrics">CNVMetrics</a></em> to calculate metrics using the
<em>CNV status calls</em> (amplification/deletion status):</p>
<p>The key functions for each step are:</p>
<div class="horizontal-scroll"><table class="table">
<thead><tr class="header">
<th>Step</th>
<th>Function</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Data Importation</td>
<td><code><a href="https://rdrr.io/pkg/GenomicRanges/man/makeGRangesListFromDataFrame.html">GenomicRanges::makeGRangesListFromDataFrame()</a></code></td>
</tr>
<tr class="even">
<td>Metric Calculation</td>
<td><code><a href="../reference/calculateOverlapMetric.html">calculateOverlapMetric()</a></code></td>
</tr>
<tr class="odd">
<td>Metric Visualization</td>
<td><code><a href="../reference/plotOverlapMetric.html">plotOverlapMetric()</a></code></td>
</tr>
</tbody>
</table></div>
<p>The <code>package::function()</code> notation is used for functions from other packages.</p>
<div id="data-input---copy-number-file" class="section level2" number="4.1">
<h2 class="hasAnchor">
<a href="#data-input---copy-number-file" class="anchor"></a><span class="header-section-number">4.1</span> Data Input - Copy number file</h2>
<p><em>CNV status calls</em> are represented as segments with a copy number state. The
state be general, such as “amplification,” “deletion” or “neutral,” or more
specific such as TODO.</p>
<p>A basic five-column input file containing genomic position
(chromosome, start, end), sample identification and <em>CNV status calls</em> is
required. All samples that need to be analyzed together have to be combined
into one file.</p>
<p>A column named <strong>state</strong> is required. In this column, the amplified and
deleted segments must be assigned those values:</p>
<ul>
<li><strong><span style="color:darkred">AMPLIFICATION</span></strong></li>
<li><strong><span style="color:darkred">DELETION</span></strong></li>
</ul>
<p>Segments with other <strong>state</strong> values can be present in the file. However,
those segments won’t be retain for the calculation of the metrics.</p>
<div class="figure" style="text-align: center">
<span id="fig:figureCNVFile"></span>
<img src="Input_CNV_call_300ppi_v01.jpg" alt="Example of a copy number file containing CNV calls." width="100%"><p class="caption">
Figure 2: <span class="caption-title">Example of a copy number file containing CNV calls</span><br></p>
</div>
</div>
<div id="data-importation---grangeslist" class="section level2" number="4.2">
<h2 class="hasAnchor">
<a href="#data-importation---grangeslist" class="anchor"></a><span class="header-section-number">4.2</span> Data Importation - GRangesList</h2>
<p>The input format for the copy number information, as needed by the
<code><a href="../reference/calculateOverlapMetric.html">calculateOverlapMetric()</a></code> function, is a <code>GRangesList</code> object.</p>
<p>The easiest way to generate a <code>GRangesList</code> object is to first load the
copy number information into an R <code>data.frame</code> and then, use the
<code><a href="https://rdrr.io/pkg/GenomicRanges/man/makeGRangesListFromDataFrame.html">GenomicRanges::makeGRangesListFromDataFrame()</a></code> function to convert them
to a <code>GRangesList</code>.</p>
<p>For this demonstration, we consider <em>CNV status calls</em> as obtained with
<em><a href="https://github.com/KrasnitzLab/CNprep">CNprep</a></em> <span class="citation">(Belleau et al. 2020)</span>,
from ten mouse pancreatic organoids <span class="citation">(Oni et al. 2020)</span>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Load required libraries</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicRanges">GenomicRanges</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/krasnitzlab/CNVMetrics">CNVMetrics</a></span><span class="op">)</span>

<span class="co">## Load file containing CNV calls for 10 mouse organoids</span>
<span class="va">data.dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package<span class="op">=</span><span class="st">"CNVMetrics"</span><span class="op">)</span>
<span class="va">cnv.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data.dir</span>, <span class="st">"mousePairedOrganoids.txt"</span><span class="op">)</span>
<span class="va">calls</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span><span class="va">cnv.file</span>, header<span class="op">=</span><span class="cn">TRUE</span>, sep<span class="op">=</span><span class="st">"\t"</span><span class="op">)</span>

<span class="co">## The CNV status calls for all samples are present in one file</span>
<span class="co">## The 'state' column is required</span>
<span class="co">## The chromosome Y has been removed</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">calls</span><span class="op">)</span></code></pre></div>
<pre><code>##    ID  chr     start       end    log2value         state
## 1 mT3 chr1   3527630 195471971  0.143253158       NEUTRAL
## 2 mT3 chr2   3549740 182113224  0.008638116       NEUTRAL
## 3 mT3 chr3   3536051 160039680  0.032630258       NEUTRAL
## 4 mT3 chr4   3734693 156508116 -0.026741963       NEUTRAL
## 5 mT3 chr5   3575586 111373751 -0.018519856       NEUTRAL
## 6 mT3 chr5 111373751 114262563  0.226961493 AMPLIFICATION</code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## The ID column identifies the 10 samples</span>
<span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">calls</span><span class="op">[</span>,<span class="st">"ID"</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code>##  [1] "mT3"  "mM6"  "mM10" "mM15" "mT6"  "mT19" "mT23" "mT34" "mM1"  "mM3P"</code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## The ID column is used to split the samples into different GRanges </span>
<span class="co">## inside a GRangesList</span>
<span class="co">## The 'keep.extra.column=TRUE' parameter is needed to retained the extra </span>
<span class="co">## column 'state' that is needed for the calculation of the metrics</span>
<span class="va">grl</span> <span class="op">&lt;-</span> <span class="fu">GenomicRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/makeGRangesListFromDataFrame.html">makeGRangesListFromDataFrame</a></span><span class="op">(</span><span class="va">calls</span>, 
    split.field<span class="op">=</span><span class="st">"ID"</span>, keep.extra.columns<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="va">grl</span></code></pre></div>
<pre><code>## GRangesList object of length 10:
## $mM1
## GRanges object with 33 ranges and 2 metadata columns:
##        seqnames             ranges strand |  log2value         state
##           &lt;Rle&gt;          &lt;IRanges&gt;  &lt;Rle&gt; |  &lt;numeric&gt;   &lt;character&gt;
##    [1]     chr1   3527630-62533399      * |  0.0351701       NEUTRAL
##    [2]     chr1  62533399-82082589      * | -0.1066207       NEUTRAL
##    [3]     chr1 82082589-195471971      * |  0.0807449       NEUTRAL
##    [4]     chr2  3549740-182113224      * |  0.0527007       NEUTRAL
##    [5]     chr3  3536051-136509203      * |  0.3431661 AMPLIFICATION
##    ...      ...                ...    ... .        ...           ...
##   [29]    chr18   3558909-90702639      * |  0.0736673       NEUTRAL
##   [30]    chr19   3614043-61431566      * |  0.0902238       NEUTRAL
##   [31]     chrX   5577384-62973719      * | -0.2259987       NEUTRAL
##   [32]     chrX  62973719-84082850      * | -0.0641819       NEUTRAL
##   [33]     chrX 84082850-169538352      * | -0.3902963      DELETION
##   -------
##   seqinfo: 20 sequences from an unspecified genome; no seqlengths
## 
## ...
## &lt;9 more elements&gt;</code></pre>
</div>
<div id="metric-calculation" class="section level2" number="4.3">
<h2 class="hasAnchor">
<a href="#metric-calculation" class="anchor"></a><span class="header-section-number">4.3</span> Metric Calculation</h2>
<p>The calculation of the similarity metrics is done with the
<code><a href="../reference/calculateOverlapMetric.html">calculateOverlapMetric()</a></code> function.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">metric</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculateOverlapMetric.html">calculateOverlapMetric</a></span><span class="op">(</span><span class="va">grl</span>, method <span class="op">=</span> <span class="st">"sorensen"</span><span class="op">)</span>

<span class="va">metric</span></code></pre></div>
<pre><code>## CNV Metric done on overlapping regions
## Metric:
## sorensen
## 
## AMPLIFICATION:
##            mM1      mM10 mM15      mM3P mM6
## mM1         NA        NA   NA        NA  NA
## mM10 0.4437132        NA   NA        NA  NA
## mM15 0.0000000 0.0000000   NA        NA  NA
## mM3P 0.1713638 0.2380730    0        NA  NA
## mM6  0.4048136 0.5545147    0 0.2952499  NA
## [ -- omitted 5 rows/columns ]
## 
## DELETION:
##            mM1      mM10      mM15       mM3P mM6
## mM1         NA        NA        NA         NA  NA
## mM10 0.0000000        NA        NA         NA  NA
## mM15 0.4595765 0.0000000        NA         NA  NA
## mM3P 0.2568018 0.3336979 0.6481145         NA  NA
## mM6  0.0000000 0.0000000 0.0000000 0.00918594  NA
## [ -- omitted 5 rows/columns ]</code></pre>
</div>
<div id="metric-visualization" class="section level2" number="4.4">
<h2 class="hasAnchor">
<a href="#metric-visualization" class="anchor"></a><span class="header-section-number">4.4</span> Metric Visualization</h2>
<p>A heatmap of this similarity metrics can be a useful tool to get an overview
over similarities and dissimilarities between samples.</p>
<p>The <code><a href="../reference/plotOverlapMetric.html">plotOverlapMetric()</a></code> function generates a graphical representation of
the similarity metrics in the form of a sample-to-sample heatmap. By default,
an hierarchical clustering based on based on the sample distances
(1-metrics) is used.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Create graph for the metrics related to amplified regions</span>
<span class="fu"><a href="../reference/plotOverlapMetric.html">plotOverlapMetric</a></span><span class="op">(</span><span class="va">metric</span>, type <span class="op">=</span> <span class="st">"AMPLIFICATION"</span><span class="op">)</span></code></pre></div>
<p><img src="CNVMetrics_files/figure-html/demoPlot01-1.png" width="100%" style="display: block; margin: auto;"></p>
<p><br></p>
<p><br></p>
<p>The <code><a href="../reference/plotOverlapMetric.html">plotOverlapMetric()</a></code> function uses the <em><a href="https://CRAN.R-project.org/package=pheatmap">pheatmap</a></em> package to
generate the graph. All arguments accepted by <code><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html">pheatmap::pheatmap</a></code> function
are valid arguments.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Create graph for the metrics related to deleted regions</span>
<span class="co">## Metric value is printed as 'display_numbers' and 'number_format' are</span>
<span class="co">## arguments recognized by pheatmap() function</span>
<span class="fu"><a href="../reference/plotOverlapMetric.html">plotOverlapMetric</a></span><span class="op">(</span><span class="va">metric</span>, type <span class="op">=</span> <span class="st">"DELETION"</span>, 
                    colorRange<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"white"</span>, <span class="st">"darkorange"</span><span class="op">)</span>,
                    show_colnames<span class="op">=</span><span class="cn">TRUE</span>,
                    display_numbers<span class="op">=</span><span class="cn">TRUE</span>,
                    number_format<span class="op">=</span><span class="st">"%.2f"</span><span class="op">)</span></code></pre></div>
<p><img src="CNVMetrics_files/figure-html/demoPlot02-1.png" width="100%" style="display: block; margin: auto;"></p>
<p><br></p>
<p><br></p>
<p>Row and/or column annotation is often useful and can easy by included with
the <code>annotation_row</code> or <code>annotation_col</code> arguments, as described in the
<code><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html">pheatmap::pheatmap</a></code> method.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Load file containing annotations for the mouse organoids</span>
<span class="co">## The mouse ID identifying the source of the sample</span>
<span class="co">## The stage identifying the state (tumor vs metastasis) of the sample</span>
<span class="va">data.dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package<span class="op">=</span><span class="st">"CNVMetrics"</span><span class="op">)</span>
<span class="va">annotation.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">data.dir</span>, <span class="st">"mousePairedOrganoidsInfo.txt"</span><span class="op">)</span>
<span class="va">annotOrg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span><span class="va">annotation.file</span>, header<span class="op">=</span><span class="cn">TRUE</span>, sep<span class="op">=</span><span class="st">"\t"</span><span class="op">)</span>

<span class="co">## The row names must correspond to the names assigned to the rows/columns</span>
<span class="co">## in the CNVMetric object</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">annotOrg</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">annotOrg</span><span class="op">$</span><span class="va">ID</span>
<span class="va">annotOrg</span><span class="op">$</span><span class="va">ID</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>

<span class="fu"><a href="https://rdrr.io/r/base/all.html">all</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">annotOrg</span><span class="op">)</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">metric</span><span class="op">$</span><span class="va">AMPLIFICATION</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Create graph for the metrics related to amplified regions</span>
<span class="co">## Rows are annotated with the stage and mouse information</span>
<span class="fu"><a href="../reference/plotOverlapMetric.html">plotOverlapMetric</a></span><span class="op">(</span><span class="va">metric</span>, type <span class="op">=</span> <span class="st">"AMPLIFICATION"</span>, 
                    colorRange<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"white"</span>, <span class="st">"steelblue"</span><span class="op">)</span>,
                    annotation_row<span class="op">=</span><span class="va">annotOrg</span><span class="op">)</span></code></pre></div>
<p><img src="CNVMetrics_files/figure-html/demoPlot03-1.png" width="100%" style="display: block; margin: auto;"></p>
<p><br></p>
</div>
</div>
<div id="metrics-using-the-cnv-status-calls" class="section level1" number="5">
<h1 class="hasAnchor">
<a href="#metrics-using-the-cnv-status-calls" class="anchor"></a><span class="header-section-number">5</span> Metrics using the <em>CNV status calls</em>
</h1>
<p>This survey represents the overlap metrics that are implemented in
<em><a href="https://github.com/KrasnitzLab/CNVMetrics">CNVMetrics</a></em> package. Those metrics are calculated
using the <em>CNV status calls</em>. The size of the amplified/deleted regions as
well as the size of the overlapping of regions are always in base paired.</p>
<div id="sørensen" class="section level2" number="5.1">
<h2 class="hasAnchor">
<a href="#s%C3%B8rensen" class="anchor"></a><span class="header-section-number">5.1</span> Sørensen</h2>
<p>The Sørensen coefficient <span class="citation">(Sørensen 1948)</span> is calculated by dividing twice the
size of the intersection by the sum of the size of the two sets:</p>
<p><span class="math display" id="eq:sorensen">\[\begin{equation}
  \frac{2\times  \left| X \cap Y \right| }{\left| X \right| + \left| Y \right|}
  \tag{1}
\end{equation}\]</span></p>
<p>where <span class="math inline">\(X\)</span> and <span class="math inline">\(Y\)</span> represent the regions of each sample in base paired.</p>
</div>
<div id="szymkiewiczsimpson" class="section level2" number="5.2">
<h2 class="hasAnchor">
<a href="#szymkiewiczsimpson" class="anchor"></a><span class="header-section-number">5.2</span> Szymkiewicz–Simpson</h2>
<p>The Szymkiewicz–Simpson coefficient <span class="citation">(Vijaymeena and K. 2016)</span>, also known as the
overlap coefficient, is calculated by dividing the size of the intersection
by the smaller of the size of the two sets:</p>
<p><span class="math display" id="eq:szymkiewicz">\[\begin{equation}
  \frac{\left| X \cap Y \right|}{min \left(\left| X \right|,\left| Y \right|\right)}
  \tag{2}
\end{equation}\]</span></p>
<p>where <span class="math inline">\(X\)</span> and <span class="math inline">\(Y\)</span> represent the regions of each sample in base paired. If
set <span class="math inline">\(X\)</span> is a subset of <span class="math inline">\(Y\)</span> or vice versa, the overlap coefficient
value is 1.</p>
</div>
<div id="jaccard" class="section level2" number="5.3">
<h2 class="hasAnchor">
<a href="#jaccard" class="anchor"></a><span class="header-section-number">5.3</span> Jaccard</h2>
<p>The Jaccard coefficient <span class="citation">(Jaccard 1912)</span>, also known as coefficient of
community, is calculated by dividing the size
of the intersection by the smaller of the size of the two sets:</p>
<p><span class="math display" id="eq:jaccard">\[\begin{equation}
  \frac{\left| X \cap Y \right| }{ \left| X \cup Y \right|}
  \tag{3}
\end{equation}\]</span></p>
<p>where <span class="math inline">\(X\)</span> and <span class="math inline">\(Y\)</span> represent the regions of each sample in base paired.</p>
</div>
</div>
<div id="metrics-using-the-level-of-amplificationdeletion" class="section level1" number="6">
<h1 class="hasAnchor">
<a href="#metrics-using-the-level-of-amplificationdeletion" class="anchor"></a><span class="header-section-number">6</span> Metrics using the level of amplification/deletion</h1>
<p>This survey represents the similarity measures that are implemented in
<em><a href="https://github.com/KrasnitzLab/CNVMetrics">CNVMetrics</a></em> package. Those metrics are calculated
using the level of amplification/deletion. The level of
amplification/deletion is in log2 ratio while the size of the regions
is in base paired.</p>
<div id="weighted-euclidean-distance" class="section level2" number="6.1">
<h2 class="hasAnchor">
<a href="#weighted-euclidean-distance" class="anchor"></a><span class="header-section-number">6.1</span> Weighted Euclidean Distance</h2>
<p>The Weighted Euclidean Distance corresponds to the euclidean distance between
the log2 values of the two samples multiplied by the natural logarithm
of the number of bases of the analyzed segments. The final metric is the
squared sum of the values obtained for all segments included in the
calculation.</p>
<p><span class="math display" id="eq:euclidean">\[\begin{equation}
  \sqrt{\sum_{i=1} log_{2}(w_{i}) (A_{i} - B_{i})^{2}}
  \tag{4}
\end{equation}\]</span></p>
<p>where <span class="math inline">\(A_{i}\)</span> and <span class="math inline">\(B_{i}\)</span> represent the log2 ratio values of samples <span class="math inline">\(A\)</span> and
<span class="math inline">\(B\)</span> for the region <span class="math inline">\(i\)</span> while <span class="math inline">\(w_{i}\)</span> is the length of region <span class="math inline">\(i\)</span> in base
paired.</p>
</div>
</div>
<div id="supplementary-information" class="section level1" number="7">
<h1 class="hasAnchor">
<a href="#supplementary-information" class="anchor"></a><span class="header-section-number">7</span> Supplementary information</h1>
<div id="creating-your-own-grangeslist" class="section level2" number="7.1">
<h2 class="hasAnchor">
<a href="#creating-your-own-grangeslist" class="anchor"></a><span class="header-section-number">7.1</span> Creating your own GRangesList</h2>
<p>The <code><a href="https://rdrr.io/pkg/GenomicRanges/man/makeGRangesListFromDataFrame.html">GenomicRanges::makeGRangesListFromDataFrame()</a></code> function enables the
creation of a list of <code>GRangesList</code> objects from a <code>data.frame</code>. However,
<code>GRangesList</code> can also be generated and filled manually.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Construction a GRangesList()</span>
<span class="va">gr1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html">GRanges</a></span><span class="op">(</span>seqnames<span class="op">=</span><span class="st">"chr2"</span>, ranges<span class="op">=</span><span class="fu">IRanges</span><span class="op">(</span><span class="fl">3</span>, <span class="fl">6000</span><span class="op">)</span>,
          strand<span class="op">=</span><span class="st">"+"</span>, state<span class="op">=</span><span class="st">"AMPLIFICATION"</span>, 
          log2ratio<span class="op">=</span><span class="fl">0.45</span><span class="op">)</span>
<span class="va">gr2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html">GRanges</a></span><span class="op">(</span>seqnames<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"chr2"</span><span class="op">)</span>,
          ranges<span class="op">=</span><span class="fu">IRanges</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">7</span>,<span class="fl">5555</span><span class="op">)</span>, width<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1200</span>, <span class="fl">40</span><span class="op">)</span><span class="op">)</span>,
          strand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"+"</span>, <span class="st">"-"</span><span class="op">)</span>,  state<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"NEUTRAL"</span>, <span class="st">"AMPLIFICATION"</span><span class="op">)</span>, 
          log2ratio<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.034</span>, <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span>
<span class="va">gr3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html">GRanges</a></span><span class="op">(</span>seqnames<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"chr2"</span><span class="op">)</span>,
          ranges<span class="op">=</span><span class="fu">IRanges</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">5577</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">5666</span><span class="op">)</span><span class="op">)</span>,
          strand<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"-"</span>, <span class="st">"-"</span><span class="op">)</span>, state<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"NEUTRAL"</span>, <span class="st">"AMPLIFICATION"</span><span class="op">)</span>, 
          log2ratio<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.04</span>, <span class="fl">0.31</span><span class="op">)</span><span class="op">)</span>

<span class="va">grl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRangesList-class.html">GRangesList</a></span><span class="op">(</span><span class="st">"sample01"</span><span class="op">=</span><span class="va">gr1</span>, <span class="st">"sample02"</span><span class="op">=</span><span class="va">gr2</span>, <span class="st">"sample03"</span><span class="op">=</span><span class="va">gr3</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="session-info" class="section level1" number="8">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a><span class="header-section-number">8</span> Session info</h1>
<p>Here is the output of <code><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo()</a></code> on the system on which this document was
compiled:</p>
<pre><code>## R version 4.0.3 (2020-10-10)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04 LTS
## 
## Matrix products: default
## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats4    parallel  stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
## [1] CNVMetrics_0.1.4     GenomicRanges_1.42.0 GenomeInfoDb_1.26.7 
## [4] IRanges_2.24.1       S4Vectors_0.28.1     BiocGenerics_0.36.1 
## [7] knitr_1.31           BiocStyle_2.18.1    
## 
## loaded via a namespace (and not attached):
##  [1] highr_0.8                   compiler_4.0.3             
##  [3] BiocManager_1.30.10         RColorBrewer_1.1-2         
##  [5] XVector_0.30.0              MatrixGenerics_1.2.1       
##  [7] bitops_1.0-6                tools_4.0.3                
##  [9] zlibbioc_1.36.0             digest_0.6.27              
## [11] lattice_0.20-41             evaluate_0.14              
## [13] lifecycle_1.0.0             gtable_0.3.0               
## [15] rlang_0.4.10                Matrix_1.3-2               
## [17] DelayedArray_0.16.3         yaml_2.2.1                 
## [19] xfun_0.21                   gridExtra_2.3              
## [21] GenomeInfoDbData_1.2.4      rtracklayer_1.50.0         
## [23] stringr_1.4.0               Biostrings_2.58.0          
## [25] grid_4.0.3                  Biobase_2.50.0             
## [27] R6_2.5.0                    BiocParallel_1.24.1        
## [29] XML_3.99-0.5                rmarkdown_2.6              
## [31] bookdown_0.21               pheatmap_1.0.12            
## [33] farver_2.0.3                magrittr_2.0.1             
## [35] matrixStats_0.58.0          GenomicAlignments_1.26.0   
## [37] Rsamtools_2.6.0             scales_1.1.1               
## [39] htmltools_0.5.1.1           SummarizedExperiment_1.20.0
## [41] colorspace_2.0-0            stringi_1.5.3              
## [43] RCurl_1.98-1.2              munsell_0.5.0              
## [45] crayon_1.4.1</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Belleau2020" class="csl-entry">
Belleau, P, A Deschênes, G Sun, David A Tuveson, and Alex Krasnitz. 2020. <span>“<span class="nocase">CNprep: Copy number event detection [version 1; not peer reviewed]</span>.”</span> <em>F1000Research</em> 9 (743): (poster). <a href="https://doi.org/10.7490/f1000research.1118065.1">https://doi.org/10.7490/f1000research.1118065.1</a>.
</div>
<div id="ref-Gendoo2019" class="csl-entry">
Gendoo, Deena M. A., Robert E. Denroche, Amy Zhang, Nikolina Radulovich, Gun Ho Jang, Mathieu Lemire, Sandra Fischer, et al. 2019. <span>“<span class="nocase">Whole genomes define concordance of matched primary, xenograft, and organoid models of pancreas cancer</span>.”</span> Edited by Amos Tanay. <em>PLOS Computational Biology</em> 15 (1): e1006596. <a href="https://doi.org/10.1371/journal.pcbi.1006596">https://doi.org/10.1371/journal.pcbi.1006596</a>.
</div>
<div id="ref-Gu2016a" class="csl-entry">
Gu, Zuguang, Roland Eils, and Matthias Schlesner. 2016. <span>“<span class="nocase">gtrellis: an R/Bioconductor package for making genome-level Trellis graphics</span>.”</span> <em>BMC Bioinformatics</em> 17 (1): 169. <a href="https://doi.org/10.1186/s12859-016-1051-4">https://doi.org/10.1186/s12859-016-1051-4</a>.
</div>
<div id="ref-Hiroi2013" class="csl-entry">
Hiroi, N, T Takahashi, A Hishimoto, T Izumi, S Boku, and T Hiramoto. 2013. <span>“<span class="nocase">Copy number variation at 22q11.2: from rare variants to common mechanisms of developmental neuropsychiatric disorders</span>.”</span> <em>Molecular Psychiatry</em> 18 (11): 1153–65. <a href="https://doi.org/10.1038/mp.2013.92">https://doi.org/10.1038/mp.2013.92</a>.
</div>
<div id="ref-Jaccard1912" class="csl-entry">
Jaccard, Paul. 1912. <span>“<span class="nocase">The distribution of the flora in the alpine zone.</span>”</span> <em>New Phytologist</em> 11 (2): 37–50. <a href="https://doi.org/10.1111/j.1469-8137.1912.tb05611.x">https://doi.org/10.1111/j.1469-8137.1912.tb05611.x</a>.
</div>
<div id="ref-Oni2020" class="csl-entry">
Oni, Tobiloba E, Giulia Biffi, Lindsey A Baker, Yuan Hao, Claudia Tonelli, Tim D D Somerville, Astrid Deschênes, et al. 2020. <span>“<span class="nocase">SOAT1 promotes mevalonate pathway dependency in pancreatic cancer</span>.”</span> <em>Journal of Experimental Medicine</em> 217 (9). <a href="https://doi.org/10.1084/jem.20192389">https://doi.org/10.1084/jem.20192389</a>.
</div>
<div id="ref-Stratton2009" class="csl-entry">
Stratton, Michael R, Peter J Campbell, and P Andrew Futreal. 2009. <span>“<span class="nocase">The cancer genome</span>.”</span> <em>Nature</em> 458 (7239): 719–24. <a href="https://doi.org/10.1038/nature07943">https://doi.org/10.1038/nature07943</a>.
</div>
<div id="ref-Sorensen48" class="csl-entry">
Sørensen, Thorvald. 1948. <span>“<span class="nocase">A method of establishing groups of equal amplitude in plant sociology based on similarity of species and its application to analyses of the vegetation on Danish commons</span>.”</span> <em>Biologiske Skrifter</em> 29 (5): 1–34.
</div>
<div id="ref-Vijaymeena2016" class="csl-entry">
Vijaymeena, M. K, and Kavitha K. 2016. <span>“<span class="nocase">A Survey on Similarity Measures in Text Mining</span>.”</span> <em>Machine Learning and Applications: An International Journal</em> 3 (1): 19–28. <a href="https://doi.org/10.5121/mlaij.2016.3103">https://doi.org/10.5121/mlaij.2016.3103</a>.
</div>
<div id="ref-Zhao2013" class="csl-entry">
Zhao, Min, Qingguo Wang, Quan Wang, Peilin Jia, and Zhongming Zhao. 2013. <span>“<span class="nocase">Computational tools for copy number variation (CNV) detection using next-generation sequencing data: features and perspectives</span>.”</span> <em>BMC Bioinformatics</em> 14 (S11): S1. <a href="https://doi.org/10.1186/1471-2105-14-S11-S1">https://doi.org/10.1186/1471-2105-14-S11-S1</a>.
</div>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script><!-- tabsets --><script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script><!-- code folding --><script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": {
      styles: {
        ".MathJax_Display": {
           "text-align": "center",
           padding: "0px 150px 0px 65px",
           margin: "0px 0px 0.5em"
        },
        "@media screen and (max-width: 991px)": {
            ".MathJax_Display": {
               "text-align": "center",
               padding: "0 0 0 0"
            }
         }
      }
    }
  });
</script>
</body>
</html>
